packages/vite/src/node/plugins/importAnalysisBuild.ts

```diff
@@ -157,6 +152,18 @@ export function buildImportAnalysisPlugin(config: ResolvedConfig): Plugin {
           source.slice(start, end) === 'import.meta' &&
           source.slice(end, end + 5) === '.glob'
         ) {
+          // es worker allow globEager / glob
+          // iife worker just allow globEager
+          if (
+            isWorker &&
+            config.worker.format === 'iife' &&
+            source.slice(end, end + 10) !== '.globEager'
+          ) {
+            this.error(
+              'can not use dynamic import(import.meta.glob) in iife worker. Just support import.meta.globEager in `format: iife` worker.',
```

Comment:
```suggestion
              '`import.meta.glob` is not supported in workers with `format: iife`, use `import.meta.globEager` instead.',
```

nit on error message.